tests/pthread/test_pthread_atexit.c

```diff
@@ -0,0 +1,44 @@
+#include <assert.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+pthread_mutex_t count_lock = PTHREAD_MUTEX_INITIALIZER;
+pthread_cond_t count_nonzero = PTHREAD_COND_INITIALIZER;
+unsigned count;
+
+pthread_t thread;
+
+void *workerThread(void* arg) {
+  pthread_mutex_lock(&count_lock);
+  while (count == 0)
+    pthread_cond_wait(&count_nonzero, &count_lock);
+  count--;
+  pthread_mutex_unlock(&count_lock);
+
+  return NULL;
+}
+
+void terminateThread() {
+  pthread_mutex_lock(&count_lock);
+  if (count == 0)
+    pthread_cond_signal(&count_nonzero);
+  count++;
```

Comment:
I think this can be simplified to just:

```
count = 1;
pthread_cond_signal(&count_nonzero);
```

I might even just rename `int count` to `bool should_exit` since we don't need to actually count anything.

Reply:
Indeed, counting isn't necessary here (there are not multiple threads). Simplified with commit 4f5be19df19e2967642e93d3aaf1ec8369d005d4.